File: CS/Dashboard_CustomExtractDriver/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DataAccess.Excel;
using DevExpress.XtraEditors;
using System;

namespace Dashboard_CustomExtractDriver
{
    public partial class Form1 : XtraForm {
        const string extractFileNamePattern = "\"Extract_\"yyyyMMddHHmmssfff\".dat\"";
        public Form1()
        {
            InitializeComponent();
            ExtractDriverStorage.DefaultDriver = new ExtractEncryptionDriver();

            DashboardExtractDataSource extractDataSource = new DashboardExtractDataSource();
            extractDataSource.ExtractSourceOptions.DataSource = CreateExcelDataSource();
            extractDataSource.FileName = DateTime.Now.ToString(extractFileNamePattern);
            extractDataSource.UpdateExtractFile();

            dashboardViewer1.Dashboard = CreateDashboard(extractDataSource);
        }

        private static Dashboard CreateDashboard(DashboardExtractDataSource extractDataSource)
        {
            Dashboard dashboard = new Dashboard();
            dashboard.DataSources.Add(extractDataSource);
            PivotDashboardItem pivot = new PivotDashboardItem();
            pivot.DataSource = extractDataSource;
            pivot.Values.AddRange(new Measure("Extended Price"), new Measure("Quantity"));
            pivot.Columns.Add(new Dimension("OrderDate", DateTimeGroupInterval.Year));
            pivot.Rows.Add(new Dimension("ProductName"));
            dashboard.Items.Add(pivot);
            return dashboard;
        }

        private static DashboardExcelDataSource CreateExcelDataSource()
        {
            DashboardExcelDataSource excelDataSource = new DashboardExcelDataSource()
            {
                FileName = @"Data\SalesPerson.xlsx",
                SourceOptions = new ExcelSourceOptions()
                {
                    ImportSettings = new ExcelWorksheetSettings("Data")
                }
            };
            excelDataSource.Fill();
            return excelDataSource;
        }
    }
}

File: VB/Dashboard_CustomExtractDriver/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DataAccess.Excel
Imports DevExpress.XtraEditors
Imports System

Namespace Dashboard_CustomExtractDriver
	Partial Public Class Form1
		Inherits XtraForm

		Private Const extractFileNamePattern As String = """Extract_""yyyyMMddHHmmssfff"".dat"""
		Public Sub New()
			InitializeComponent()
			ExtractDriverStorage.DefaultDriver = New ExtractEncryptionDriver()

			Dim extractDataSource As New DashboardExtractDataSource()
			extractDataSource.ExtractSourceOptions.DataSource = CreateExcelDataSource()
			extractDataSource.FileName = Date.Now.ToString(extractFileNamePattern)
			extractDataSource.UpdateExtractFile()

			dashboardViewer1.Dashboard = CreateDashboard(extractDataSource)
		End Sub

		Private Shared Function CreateDashboard(ByVal extractDataSource As DashboardExtractDataSource) As Dashboard
			Dim dashboard As New Dashboard()
			dashboard.DataSources.Add(extractDataSource)
			Dim pivot As New PivotDashboardItem()
			pivot.DataSource = extractDataSource
			pivot.Values.AddRange(New Measure("Extended Price"), New Measure("Quantity"))
			pivot.Columns.Add(New Dimension("OrderDate", DateTimeGroupInterval.Year))
			pivot.Rows.Add(New Dimension("ProductName"))
			dashboard.Items.Add(pivot)
			Return dashboard
		End Function

		Private Shared Function CreateExcelDataSource() As DashboardExcelDataSource
			Dim excelDataSource As New DashboardExcelDataSource() With {
				.FileName = "Data\SalesPerson.xlsx", .SourceOptions = New ExcelSourceOptions() With {.ImportSettings = New ExcelWorksheetSettings("Data")}
			}
			excelDataSource.Fill()
			Return excelDataSource
		End Function
	End Class
End Namespace
